reana_commons/gherkin_parser/parser.py

```diff
@@ -0,0 +1,266 @@
+                step_mapping[step_type][step_text]["arguments"] = result.named
+                found = True
+                break
+        if not found:
+            logging.error(f"No step definition found for step: {step_text}")
+            raise StepDefinitionNotFound(
+                f"No step definition found for step: {step_text}"
+            )
+    return step_mapping
+
+
+def run_tests(
+    analysis_run_id: str,
+    workflow: str,
+    feature_name: str,
+    feature_file,
+    step_mapping: Dict,
+) -> List[TestResult]:
+    """Run all the tests in the parsed feature file.
+
+    :param feature_name: The name of the feature inside the feature file.
+    :param analysis_run_id:
+    :param workflow: The name of the workflow in REANA
+    :param feature_file: The parsed and compiled feature file, as returned by `parse_feature_file`.
+    :param step_mapping: A dictionary mapping each step to the corresponding step definition (function).
+    :return: A list of dictionaries, each corresponding to one scenario of the feature file.
+    """
+    test_results = []
+    for scenario in feature_file:
+        logging.info(f"Running scenario: {scenario['name']}...")
+        result = AnalysisTestStatus.passed
+        failed_testcase = None
+        error_log = None
+        for step in scenario["steps"]:
+            logging.debug(f"Running step: {step['text']} ({step['type']})...")
+            step_type = step["type"].strip()
+            step_text = _get_step_text(step)
+            function = step_mapping[step_type].get(step_text).get("function")
+            arguments = step_mapping[step_type].get(step_text).get("arguments")
+            if function is not None:
+                try:
+                    function(workflow, **arguments)
+                except StepSkipped:
+                    logging.info(f"Scenario skipped! Current testcase: {step_text}")
+                    result = AnalysisTestStatus.skipped
+                    break
+                except Exception as e:
+                    # Catches all exceptions, including AssertionError.
+                    result = AnalysisTestStatus.failed
+                    failed_testcase = step_text
+                    error_log = str(e)
+                    logging.error(f"Scenario failed! Failed testcase: {step_text}")
+                    logging.error(f"Error log: {e}")
+                    break
+        test_results.append(
+            TestResult(
+                scenario=scenario["name"],
+                failed_testcase=failed_testcase,
+                result=result,
+                error_log=error_log,
+                analysis_run_id=analysis_run_id,
+                feature=feature_name,
+                checked_at=datetime.now(),
+            )
+        )
+        if result == AnalysisTestStatus.passed:
+            logging.info(f"Scenario `{scenario['name']}` passed!")
+
+    return test_results
+
+
+def parse_and_run_tests(
+    analysis_run_id: str,
```

Comment:
minor: `analysis_run_id` is needed only in continuous-reuse, while it's not needed in reana-commons/reana-client. It identifies the analysis run, which is not the same as the workflow. The workflow details are fetched from REANA's API using `workflow` as identifier (either name or ID). I guess it can be removed from here and elsewhere in commons/client?

Reply:
We can also skip doing this if too many things need to be changed